packages/core/strapi/lib/commands/start.js

```diff
@@ -9,11 +10,8 @@ module.exports = async specifiedDir => {
   const appDir = process.cwd();
   const isTSProject = await tsUtils.isUsingTypeScript(appDir);
   const outDir = await tsUtils.resolveOutDir(appDir);
-  if (isTSProject)
-    await tsUtils.compile(appDir, {
-      watch: false,
-      configOptions: { options: { incremental: true } },
-    });
+  const buildDirExists = fs.existsSync(outDir);
+  if (isTSProject && !buildDirExists) throw new Error('Please build first to run this command');
```

Comment:
What do you think of `${outDir}" directory not found. Please run the build command before starting your application`

Reply:
sounds good will implement it